Content.Server/GameObjects/EntitySystems/MoverSystem.cs

```diff
@@ -149,7 +157,27 @@ private void UpdateKinematics(ITransformComponent transform, PlayerInputMoverCom
                 if (mover.StepSoundDistance > distanceNeeded)
                 {
                     mover.StepSoundDistance = 0;
-                    PlayFootstepSound(transform.GridPosition);
+                    var isFootstepModified = false;
+                    if (mover.Owner.TryGetComponent<InventoryComponent>(out var inventory)
+                        && inventory.TryGetSlotItem<ItemComponent>(EquipmentSlotDefines.Slots.SHOES, out var item))
+                    {
+                        var footstepmods = item.Owner.GetAllComponents<IOnMove>().ToList();
```

Comment:
There is a very big distinction between "on move" and what this clearly turned out to be.

Just make this do a `TryGetComponent(out FootstepModifierComponent modifier)` here. I very much doubt you need to make this more complicated.